Looms.PoS/Looms.PoS.Application/Features/Auth/Commands/Login/LoginRequestValidator.cs

```diff
@@ -0,0 +1,40 @@
+﻿using FluentValidation;
+using Looms.PoS.Application.Models.Requests.Auth;
+using Looms.PoS.Domain.Exceptions;
+using Looms.PoS.Domain.Interfaces;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Looms.PoS.Application.Features.Auth.Commands.Login;
+
+public class LoginRequestValidator : AbstractValidator<LoginRequest>
+{
+    public LoginRequestValidator(IUsersRepository usersRepository)
+    {
+        RuleFor(x => x.Email)
+            .Cascade(CascadeMode.Stop)
+            .EmailAddress()
+            .MustAsync(async (email, cancellationToken) =>
+            {
+                if (!await usersRepository.ExistsWithEmail(email))
+                {
+                    throw new LoomsUnauthorizedException("Invalid credentials");
+                }
+                return true;
+            })
```

Comment:
You could just try catch when getting user by email, additional query for nothing

Reply:
It is better to avoid try catch when possible

Reply:
I still think it's better than making additional request to db

Reply:
What additional request?

Reply:
To check if user exists with email

Reply:
After this request is made the user is cached and GetByEmail uses the cached value

Reply:
Is is simpler to read instead of try-catch with more nesting inside and also executes faster

Reply:
No, it doesn't get cached

Reply:
added cache logic in GetByEmailAsync method